scripts/linters/pre_commit_linter.py

```diff
@@ -497,11 +497,21 @@ def main(args=None):
 
     all_messages = []
 
+    # Prepare semaphore for locking mechanism.
+    semaphore_max_concurrent_runs = 1
+    semaphore_concurrent_count = min(
+        multiprocessing.cpu_count(), semaphore_max_concurrent_runs)
+    semaphore = threading.Semaphore(semaphore_concurrent_count)
+
     for task in tasks_custom:
+        semaphore.acquire()
         all_messages += task.output
+        semaphore.release()
```

Comment:
sorry, I forgot to mention this in the last round, but wrap these three lines in a function. With semaphores, you want to encapsulate it well since those executed in a semi-isolated environment.

Reply:
Done.